py-polars/tests/unit/io/test_parquet.py

```diff
@@ -784,3 +785,33 @@ def test_parquet_array_statistics() -> None:
     assert pl.scan_parquet("test.parquet").filter(
         pl.col("a") != [1, 2, 3]
     ).collect().to_dict(as_series=False) == {"a": [[4, 5, 6], [7, 8, 9]], "b": [2, 3]}
+
+
+@pytest.mark.write_disk()
+def test_read_parquet_only_loads_selected_columns_15098(
+    memory_usage_without_pyarrow: MemoryUsage, tmp_path: Path
+) -> None:
+    """Only requested columns are loaded by ``read_parquet()``."""
+    tmp_path.mkdir(exist_ok=True)
+
+    # Each column will be about 8MB of RAM
+    series = pl.arange(0, 1_000_000, dtype=pl.Int64, eager=True)
+
+    file_path = tmp_path / "multicolumn.parquet"
+    df = pl.DataFrame(
+        {
+            "a": series,
+            "b": series,
+        }
+    )
+    df.write_parquet(file_path)
+    del df, series
+
+    memory_usage_without_pyarrow.reset_tracking()
+
+    # Only load one column:
+    df = pl.read_parquet([file_path], columns=["b"], rechunk=False)
+    del df
+    # Only one column's worth of memory should be used; 2 columns would be
+    # 16_000_000 at least, but there's some overhead.
+    assert 8_000_000 < memory_usage_without_pyarrow.get_peak() < 13_000_000
```

Comment:
Is there any more principled way to account for the overhead when doing assertions like this? It's fine for now, but it feels a bit unstable.

Reply:
In this case I think I observed 2MB overhead, so I just added a bit more to reduce chances of being too brittle. And ... it's difficult to account for that, so usually you just do a fudge factor. 

The good news is that it's not as bad as _time_ fudge factors, because memory allocations are much more deterministic than elapsed time. So it's a lot less fragile than tests that rely on `sleep()`ing.

Also note if this does start failing with 14MB allocation, it's maybe not a bug about column loading but could well indicate some excessive allocation, so a failing test might actually be informative.

Reply:
All right, let's leave it like this for now. If we write more tests like this one, we might need a 'default fudge factor' of sorts. But we can figure it out then.